components/dashboard/components/Recommendations/index.js

```diff
@@ -2,20 +2,37 @@
 // See LICENSE.txt for license information.
 
 import {connect} from 'react-redux';
-import {Datasets} from 'mattermost-redux/constants/user_analytics';
-import {getCurrentTeamId} from 'mattermost-redux/selectors/entities/teams';
 import {getCurrentUserId} from 'mattermost-redux/selectors/entities/users';
 
+import {updateRecommendations} from 'actions/views/dashboard';
+import {
+    getUserLearningGroups,
+    getRecommendations,
+} from 'selectors/views/dashboard';
+
 import {RecommendationsView} from './RecommendationsView';
 
+/* Empty array for when there are no learning groups. We don't want every state
+ * update to create a different empty array, because that will look like the
+ * userLearningGroup prop has changed when it actually hasn't.
+ */
+const noLearningGroups = [];
```

Comment:
The key is that `[]` is a _reference_ to an empty array. My understanding is that `mapStateToProps` is called every time state changes to determine whether the changes change the props of this component. Since the old line `12` will produce a reference to a _new_ empty array every time it is evaluated the user analytics don't have a learning group, this means the value of the `userLearningGroups` prop will change to be a different reference to a different empty list every time `mapStateToProps` is called. By using a constant instead of creating a new empty list, the reference is left unchanged.

Reply:
Hmm, that's very interesting, and would be hard to catch! I guess an alternative would be to just have the empty array in the redux state, and not create a new empty array each time. But I think the array is created dynamically in redux, so this is the only option. Yeah, really nice catch Mike,